client/landing/stepper/declarative-flow/internals/components/step-route/test/index.tsx

```diff
@@ -0,0 +1,175 @@
+	const originalLocation = window.location;
+
+	beforeAll( () => {
+		Object.defineProperty( window, 'location', {
+			value: { ...originalLocation, assign: jest.fn() },
+		} );
+		( useIntent as jest.Mock ).mockReturnValue( 'build' );
+		( useSelectedDesign as jest.Mock ).mockReturnValue( { design_type: 'premium' } );
+	} );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	afterAll( () => {
+		Object.defineProperty( window, 'location', originalLocation );
+	} );
+
+	it( 'renders the step content', async () => {
+		const { getByText } = render( { step: RegularStep } );
+
+		await waitFor( () => expect( getByText( 'Step Content' ) ).toBeInTheDocument() );
+	} );
+
+	describe( 'requiresLoggedInUser', () => {
+		it( 'redirects users to the login page if the step requires a logged in user', async () => {
+			render( { step: RequiresLoginStep } );
+
+			await waitFor( () =>
+				expect( window.location.assign ).toHaveBeenCalledWith(
+					'/start/account/user-social?variationName=some-flow&redirect_to=%2F&toStepper=true'
+				)
+			);
+		} );
+
+		it( 'renders the step content if the user is logged in', async () => {
+			( isUserLoggedIn as jest.Mock ).mockReturnValue( true );
+			const { getByText } = render( { step: RequiresLoginStep } );
+
+			await waitFor( () => expect( getByText( 'Step Content' ) ).toBeInTheDocument() );
+		} );
+	} );
+
+	describe( 'tracking', () => {
+		it( 'records a page view when the step is rendered', async () => {
+			render( { step: RegularStep } );
+
+			expect( recordPageView ).toHaveBeenCalledWith( '/', 'Setup > some-flow > some-step-slug' );
+		} );
+
+		it( 'records recordStepStart when the step is rendered', async () => {
+			render( { step: RegularStep } );
+
+			expect( recordStepStart ).toHaveBeenCalledWith( 'some-flow', 'some-step-slug', {
+				intent: 'build',
+				assembler_source: 'premium',
+				is_in_hosting_flow: false,
+			} );
+		} );
+
+		it( 'does not record start and page view when the login is required and the user is not logged in', async () => {
+			( isUserLoggedIn as jest.Mock ).mockReturnValue( false );
+			render( { step: RequiresLoginStep } );
+
+			expect( recordStepStart ).not.toHaveBeenCalled();
+			expect( recordPageView ).not.toHaveBeenCalled();
+		} );
+
+		it( 'skips tracking when the step is re-entered', () => {
+			( getSignupCompleteFlowNameAndClear as jest.Mock ).mockReturnValue( 'some-flow' );
+			( getSignupCompleteStepNameAndClear as jest.Mock ).mockReturnValue( 'some-step-slug' );
+
+			render( { step: RegularStep } );
+
+			expect( recordStepStart ).not.toHaveBeenCalled();
+		} );
+
+		it( 'skips trackings when the renderStep returns null', () => {
+			render( { step: RegularStep, renderStep: () => null } );
+
+			expect( recordStepStart ).not.toHaveBeenCalled();
+			expect( recordPageView ).not.toHaveBeenCalled();
+		} );
+
+		it.todo( 'skips when there is no site slug or ID', () => {} );
```

Comment:
Apparently this fixes it. 

```suggestion
		it.todo( 'skips when there is no site slug or ID' );
```

Reply:
It will not be part of the final PR